blocks/edit/da-editor/da-editor.js

```diff
@@ -72,6 +84,7 @@ export default class DaEditor extends LitElement {
 
   updated(props) {
     if (!this._imsLoaded) return;
+    this.disconnectWebsocket();
```

Comment:
The way I read this is that the websocket will be disconnected for every update... and that seems a little worrisome / noisy.

Reply:
What we were seeing was that when you move to a different document, that connections to the previous document were still going on because the websocket wasn't closed. This caused errors in the log, but also (potentially a lot of) unnecessary websocket communication.

IIRC this is because the document URL didn't change ( http://localhost:3000/edit#/da-sites/da-status/bosschae/doc1 vs http://localhost:3000/edit#/da-sites/da-status/bosschae/doc2 ) is really just an anchor change, the location is still the same.

So, no the websocket won't get disconnected for every update, but it will get disconnected when the anchor is updated, i.e. we're moving to a different document.

Reply:
Actually, I think @auniverseaway is correct - this looks wrong. It should only disconnect if initProse is called. In other words, I think it needs to be moved inside the if statement one line down.

Reply:
@bosschaert, I think you latest merge did mess this up. You never assign the wsProvider and this disconnect should be in the if I think.

Reply:
@bosschaert I didn't think of the doc change... 

This makes me think we need to get a wiki of E2E test steps for our own reminders of what to test for these high impact changes.

1. Can add table.
2. Can edit doc.
3. Can edit doc with another user.
4. Can switch docs via hashchange

etc.

Ideally this would funnel into Playwright so we don't have to do this stuff manually.

Reply:
@auniverseaway, should be fixed - can you approve again?

Reply:
Thanks for fixing @karlpauls yes it got messed up in the merge.